app/components/channel_info/index.js

```diff
@@ -0,0 +1,420 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {injectIntl, intlShape} from 'react-intl';
+import {
+    Dimensions,
+    Platform,
+    TouchableWithoutFeedback,
+    View,
+    Text,
+    findNodeHandle
+} from 'react-native';
+import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
+
+import EventEmitter from 'mattermost-redux/utils/event_emitter';
+
+import ErrorText from 'app/components/error_text';
+import FormattedText from 'app/components/formatted_text';
+import Loading from 'app/components/loading';
+import StatusBar from 'app/components/status_bar';
+import TextInputWithLocalizedPlaceholder from 'app/components/text_input_with_localized_placeholder';
+
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import {getShortenedURL} from 'app/utils/url';
+
+class ChannelInfo extends PureComponent {
+    static propTypes = {
+        intl: intlShape.isRequired,
+        navigator: PropTypes.object,
+        theme: PropTypes.object.isRequired,
+        displayHeaderOnly: PropTypes.bool,
+        creating: PropTypes.bool,
+        editing: PropTypes.bool,
+        updating: PropTypes.bool,
```

Comment:
what's the difference between creating|editing|updating ? can't they be only one?

Reply:
@enahum Initially the only prop that existed was `creating`

* creating: when the user submits a new (private or public) channel this gets set to true
* editing: a flag that's used to signify it's in edit mode
* updating: when the user submits an update to the server this gets set to true.

editing has one purpose
but creating and updating are very similar just the wording changes based on the context of the form (create or edit)

we could change creating and updating to: `saving`